src/context/context.js

```diff
@@ -0,0 +1,40 @@
+import React, {Component} from 'react'
+import {db} from '../firebase.js'
+
+const UserContext = React.createContext()
+
+class UserProvider extends Component {
+  state = {
+    shoppingList: []
+  }
+
+  componentDidMount(){
+    db.collection('shoppingList').get()
+      .then(querySnapshot => {
+        const data = querySnapshot.docs.map(doc => doc.data());
+        console.log(data)
+        this.setState({shoppingList: data})
+      })
+  }
+
+  getShoppingList = () => this.state.shoppingList;
+
+  render() {
+    const { children } = this.props
+    const { getShoppingList } = this
+
+    return (
+      <UserContext.Provider
+        value={{
+          getShoppingList
+        }}
+      >
+        {children}
+      </UserContext.Provider>
```

Comment:
Love the context hook. Fantastic! I'd love to see a quick explanation of this in the demos for the team tomorrow! 

We could use the same type of context hook and provider to extract the token logic from `App.js`